client/src/core/services/parse-map.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+      providedIn:'root',
+})
+export class ParseMapService {
+    terrain: any = {
+        name: 'Terrain 1',
+        height: 30,
+        width: 50,
+        obstacles: [],
+        trees: [],
+    };
+
+        constructor() { }
+    getMap(): void {
+        return this.terrain;
+    }
+
+    parseFile(file: string): void {
+        let fileTab = file.split('\n');
+        for (const fil of fileTab) {
+            for(var _i = 0; _i < fil.length; _i++) {
+                if (fil[_i] === 'O') {
+                        const coordo = {x:Number(_i) , y:Number(fileTab.indexOf(fil)) };
+                        this.terrain.trees.push(coordo);
+                }
+                if (fil[_i] === 'X') {
+                        const coordo = {x:Number(_i) , y:Number(fileTab.indexOf(fil)) };
+                        this.terrain.obstacles.push(coordo);
```

Comment:
Codacy found an issue: [Expected indentation of 20 spaces but found 24.](https://app.codacy.com/gh/Ca-Cybersecurity-Robiot/robiot/pullRequest?prid=5943029)